doc/src/base/sort.md

```diff
@@ -165,22 +168,21 @@ Base.Sort.defalg(::AbstractArray{<:Union{SmallInlineStrings, Missing}}) = Inline
 
 ## Alternate Orderings
 
-By default, `sort` and related functions use [`isless`](@ref) to compare two
-elements in order to determine which should come first. The
-[`Base.Order.Ordering`](@ref) abstract type provides a mechanism for defining
-alternate orderings on the same set of elements: when calling a sorting function like
-`sort`, an instance of `Ordering` can be provided with the keyword argument `order`.
-
-Instances of `Ordering` define a [total order](https://en.wikipedia.org/wiki/Total_order)
-on a set of elements, so that for any elements `a`, `b`, `c` the following hold:
-
-* Exactly one of the following is true: `a` is less than `b`, `b` is less than
-  `a`, or `a` and `b` are equal (according to [`isequal`](@ref)).
-* The relation is transitive - if `a` is less than `b` and `b` is less than `c`
-  then `a` is less than `c`.
-
-The [`Base.Order.lt`](@ref) function works as a generalization of `isless` to
-test whether `a` is less than `b` according to a given order.
+By default, `sort`, `searchsorted`, and related functions use [`isless`](@ref) to compare
+two elements in order to determine which should come first. The
+[`Base.Order.Ordering`](@ref) abstract type provides a mechanism for defining alternate
+orderings on the same set of elements: when calling a sorting function like
+`sort!`, an instance of `Ordering` can be provided with the keyword argument `order`.
+
+Instances of `Ordering` define a
+[strict weak order](https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings)
+through the [`Base.Order.lt`](@ref) function, which works as
+a generalization of `isless`.
+For `lt` to be a strict weak order, for any elements `a`, `b`, `c` the following must hold:
+
+* `lt(a, b) && lt(b, a) === false`;
+* if `lt(a, b) && lt(b, c)`, then `lt(a, c)`; and
+* if `!lt(a, b) && !lt(b, c)`, then `!lt(a, c)`
```

Comment:
```suggestion
Instances of `Ordering` define an order through the [`Base.Order.lt`](@ref)
function, which works as a generalization of `isless`.
This function must satisfy all the conditions of a 
[strict weak order](https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings).
See [`sort!`](@ref) for details and examples of valid and invalid `lt` functions.
```

I think the last condition (line 185) is wrong, see https://github.com/JuliaLang/julia/pull/48363#discussion_r1085096725 for a counter-example. But anyway I propose to keep the `sort!` doc as a single reference for these intricate conditions.

Reply:
I agree that this should only be defined in one place.